pallets/ocex/src/lib.rs

```diff
@@ -0,0 +1,1021 @@
+						ingress_messages.push(
+							polkadex_primitives::ingress::IngressMessages::OpenTradingPair(
+								trading_pair.clone(),
+							),
+						);
+					});
+					Self::deposit_event(Event::OpenTradingPair { pair: trading_pair.clone() });
+				} else {
+					//scope never executed, already ensured if trading pair exits above
+				}
+			});
+			Ok(())
+		}
+
+		/// Registers a new trading pair
+		#[pallet::weight(100000)]
+		pub fn register_trading_pair(
+			origin: OriginFor<T>,
+			base: AssetId,
+			quote: AssetId,
+			min_order_price: BalanceOf<T>,
+			max_order_price: BalanceOf<T>,
+			min_order_qty: BalanceOf<T>,
+			max_order_qty: BalanceOf<T>,
+			price_tick_size: BalanceOf<T>,
+			qty_step_size: BalanceOf<T>,
+		) -> DispatchResult {
+			T::GovernanceOrigin::ensure_origin(origin)?;
+
+			ensure!(base != quote, Error::<T>::BothAssetsCannotBeSame);
+			ensure!(
+				!<TradingPairs<T>>::contains_key(base, quote),
+				Error::<T>::TradingPairAlreadyRegistered
+			);
+			ensure!(
+				!<TradingPairs<T>>::contains_key(quote, base),
+				Error::<T>::TradingPairAlreadyRegistered
+			);
+
+			// We need to also check if provided values are not zero
+			ensure!(
+				min_order_price.saturated_into::<u128>() > 0 &&
+					max_order_price.saturated_into::<u128>() > 0 &&
+					min_order_qty.saturated_into::<u128>() > 0 &&
+					max_order_qty.saturated_into::<u128>() > 0 &&
+					price_tick_size.saturated_into::<u128>() > 0 &&
+					qty_step_size.saturated_into::<u128>() > 0,
+				Error::<T>::TradingPairConfigCannotBeZero
+			);
+
+			// We need to check if the provided parameters are not exceeding 10^27 so that there
+			// will not be an overflow upon performing calculations
+			ensure!(
+				min_order_price.saturated_into::<u128>() <= DEPOSIT_MAX,
+				Error::<T>::AmountOverflow
+			);
+			ensure!(
+				max_order_price.saturated_into::<u128>() <= DEPOSIT_MAX,
+				Error::<T>::AmountOverflow
+			);
+			ensure!(
+				min_order_qty.saturated_into::<u128>() <= DEPOSIT_MAX,
+				Error::<T>::AmountOverflow
+			);
+			ensure!(
+				max_order_qty.saturated_into::<u128>() <= DEPOSIT_MAX,
+				Error::<T>::AmountOverflow
+			);
+			ensure!(
+				price_tick_size.saturated_into::<u128>() <= DEPOSIT_MAX,
+				Error::<T>::AmountOverflow
+			);
+			ensure!(
+				qty_step_size.saturated_into::<u128>() <= DEPOSIT_MAX,
+				Error::<T>::AmountOverflow
+			);
+
+			//enclave will only support min volume of 10^-8
+			//if trading pairs volume falls below it will pass a UnderFlow Error
+			ensure!(
+				min_order_price.saturated_into::<u128>() > TRADE_OPERATION_MIN_VALUE &&
+					min_order_qty.saturated_into::<u128>() > TRADE_OPERATION_MIN_VALUE &&
+					min_order_price
+						.saturated_into::<u128>()
+						.saturating_mul(min_order_qty.saturated_into::<u128>()) >
+						TRADE_OPERATION_MIN_VALUE,
+				Error::<T>::TradingPairConfigUnderflow
+			);
+
+			// TODO: Check if base and quote assets are enabled for deposits
```

Comment:
Is this todo means to check if balance has enough amount?

Reply:
It is resolved in #527